Fix duplicate node name in graph for tf.concat

yongtang opened the pull request:
This fix tries to address the issue raised in #31137 where
`Duplicate node name in graph: 'concat'` shown up with the
following:
```
import tensorflow as tf
from tensorflow.keras import Input

print('Using Tensorflow version {} (git version {})'.format(tf.version.VERSION, tf.version.GIT_VERSION))

i = Input(shape=3)
j = Input(shape=4)
try:
    print(tf.concat([i, j], axis=-1))
except Exception as e:
    print(type(e))
    print(e)
try:
    print(tf.concat([i, j], axis=-1))
except Exception as e:
    print(type(e))
    print(e)
try:
    print(tf.concat([i], axis=-1))
except Exception as e:
    print(type(e))
    print(e)
try:
    print(tf.concat(i, axis=-1))
except Exception as e:
    print(type(e))
    print(e)
```

The issue was that `identity` node passed the scope as the name.
This fix fixes the issue.

This fix fixes #31137.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

alextp commented on tensorflow/python/ops/array_ops.py:
```diff
@@ -1296,20 +1296,20 @@ def concat(values, axis, name="concat"):
       ops.convert_to_tensor(
           axis, name="concat_dim",
           dtype=dtypes.int32).get_shape().assert_has_rank(0)
-      return identity(values[0], name=scope)
```
Wouldn't a simpler fix be just to pass name=name to identity?

yongtang replied:
Thanks @alextp, indeed that should fix the issue. The PR has been updated.

yongtang committed:
Fix duplicate node name in graph for tf.concat

This fix tries to address the issue raised in 31137 where
`Duplicate node name in graph: 'concat'` shown up with the
following:
```
import tensorflow as tf
from tensorflow.keras import Input

print('Using Tensorflow version {} (git version {})'.format(tf.version.VERSION, tf.version.GIT_VERSION))

i = Input(shape=3)
j = Input(shape=4)
try:
    print(tf.concat([i, j], axis=-1))
except Exception as e:
    print(type(e))
    print(e)
try:
    print(tf.concat([i, j], axis=-1))
except Exception as e:
    print(type(e))
    print(e)
try:
    print(tf.concat([i], axis=-1))
except Exception as e:
    print(type(e))
    print(e)
try:
    print(tf.concat(i, axis=-1))
except Exception as e:
    print(type(e))
    print(e)
```

The issue was that `identity` node passed the scope as the name.
This fix fixes the issue.

This fix fixes 31137.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @alextp for the review and suggestion. The PR has been updated. Please take a look.

yongtang committed:
Fix CI build test failure.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
@alextp @rthadur one kokoro CI test was failing, because of the name change where 'concat:0' =>`concat/concat:0`. I have updated the PR and fixed the failed test.

Please take a look and sorry for the additional inconvenience.
